fix projects overview tabs

lorenzo-cavazzi opened the pull request:
This PR's primary goal is to fix the project overview stats tab (see #1074).
It shows incorrect data when:
1. the project has been recently updated.
2. the user has no specific permission for a public or internal project.

Point 1 is fixed. It happened because we never refreshed the stats. In the past, we also observed that the GitLab API was occasionally returning outdated values for a while after pushing new commits. This [should be fixed now](https://gitlab.com/gitlab-org/gitlab-foss/-/merge_requests/26854).

Point 2 is partially fixed. Sadly, the GitLab project API doesn't return statistics for a public or internal project if the user doesn't have specific permissions for that. This is unexpected because it's undocumented -- I still hope [there is another way](https://gitlab.com/gitlab-org/gitlab/-/issues/296656).  
We don't show the missing stats anymore when unavailable, but we provide extra links to the GitLab UI where some of those can be inspected.

Mind that this PR includes a few other minor changes for the overview tabs:
* Avoid fetching readme on project overview sub-tabs
* Fetch commits when opening overview/commits
* Move overview/commits components to the new ProjectsOverview files
* Merge stats into a single table (it anticipates a tiny bit of #1114 for convenience)
* Move the statistics information to the project coordinator (better to anticipate steps from #1183 whenever possible)

**Preview**: https://lorenzotest.dev.renku.ch/

**How to test**
Keep an instance of https://dev.renku.ch/ and compare the following:
* Try to move around in the project overview sub-pages on the left navigation pills. You shouldn't see queries to the readme file at each click.
* Open the project overview commits for a project and wait until the full list is fetched. Then manually navigate to another project from the projects page. If you open the commits, you should see they are now correct while wrong on dev.
* Open one of your project overview stats. Then manually navigate to a public project where you have no permission. The stats will be wrong on dev, showing the previous values.
* 
Mind that the last 2 problems may not happen if you first navigate around the project, triggering fetches from GitLab for other reasons.

fix #1074

ciyer commented on client/src/project/Project.present.js:
```diff
@@ -1507,4 +1317,4 @@ class ProjectView extends Component {
 
 // For testing
-export { filterPaths, ProjectViewCommitsBody };
+export { filterPaths };
```
Very minor, but this file should have a newline at the end.

ciyer commented on client/src/project/overview/ProjectOverview.present.js:
```diff
@@ -0,0 +1,248 @@
+    if (!metadata.accessLevel) {
+      info = (
+        <p className="font-italic">
+          Since you do not have any permission for this project, some fields may be unavailable.
```
Would it make more sense to say:
`Since you are not a member of this project, some information may not be available.`?

ciyer commented on client/src/project/overview/ProjectOverview.present.js:
```diff
@@ -0,0 +1,248 @@
+                <td className="px-3 px-lg-4 align-middle">{branches.length + 1}</td>
+                <td>
+                  <ExternalLink role="text" showLinkIcon={true}
+                    url={`${metadata.repositoryUrl}/branches`} title="Branches in Gitlab" />
```
Very minor as well (not not actually part of this PR, but I'm just noticing it, so may as well fix):
`Gitlab` -> `GitLab`

ciyer commented on client/src/project/overview/ProjectOverview.present.js:
```diff
@@ -0,0 +1,248 @@
+                <td className="px-3 px-lg-4 align-middle">{metadata.forksCount}</td>
+                <td>
+                  <ExternalLink role="text" showLinkIcon={true}
+                    url={`${metadata.repositoryUrl}/forks`} title="Forks in Gitlab" />
```
`Gitlab` -> `GitLab`

ciyer commented on client/src/project/overview/ProjectOverview.present.js:
```diff
@@ -0,0 +1,248 @@
+                <td className="px-3 px-lg-4 align-middle">{metadata.starCount}</td>
+                <td>
+                  <ExternalLink role="text" showLinkIcon={true}
+                    url={`${metadata.repositoryUrl}/-/starrers`} title="Stars in Gitlab" />
```
`Gitlab` -> `GitLab`

ciyer commented on client/src/project/overview/ProjectOverview.present.js:
```diff
@@ -0,0 +1,248 @@
+                <td className="px-3 px-lg-4 align-middle">{commitsCount}</td>
+                <td>
+                  <ExternalLink role="text" showLinkIcon={true}
+                    url={`${metadata.repositoryUrl}/commits`} title="Commits in Gitlab" />
```
`Gitlab` -> `GitLab`

ciyer commented on client/src/project/overview/ProjectOverview.present.js:
```diff
@@ -0,0 +1,248 @@
+                <td className="px-3 px-lg-4 align-middle">{storageSize}</td>
+                <td className="align-middle" rowSpan="3">
+                  <ExternalLink role="text" showLinkIcon={true}
+                    url={metadata.repositoryUrl} title="Sizes in Gitlab" />
```
`Gitlab` -> `GitLab`

ciyer reviewed (changes requested):
Code looks very good! I have some very minor things (wording, spelling of GitLab, which wasn't introduced in this PR, but I just noticed it).

I noticed some strange behavior, though.

1. Stats disappear

First look at stats shows this:
![image](https://user-images.githubusercontent.com/1196411/104196312-ec4d2780-5423-11eb-9afa-ac4ace3268e3.png)

But if I go to description and come back, I get this:
![image](https://user-images.githubusercontent.com/1196411/104196422-0ab32300-5424-11eb-93fa-529f652a0dfb.png)

2. On the above project, in RenkuLab, it says there is 1 branch, but GitLab shows 3
![image](https://user-images.githubusercontent.com/1196411/104196519-28808800-5424-11eb-9aba-500228035a69.png)

ciyer commented on client/src/project/overview/ProjectOverview.present.js:
```diff
@@ -0,0 +1,248 @@
+                </td>
+              </tr>
+              <tr>
+                <th scope="row">Storage Size</th>
```
Is there a reason to drop the old structure where Project Stats and Storage Stats where shown separately? I prefer 

![image](https://user-images.githubusercontent.com/1196411/104203201-291d1c80-542c-11eb-9947-a343762a13b4.png)

to

![image](https://user-images.githubusercontent.com/1196411/104203237-320dee00-542c-11eb-97ab-4caca59e86cb.png)

I think having two sections looks nicer and is easier to read (but I think the links in the new version are better than the buttons in the old one).

lorenzo-cavazzi replied:
As discussed offline, we will keep the separate structure for now.

lorenzo-cavazzi commented:
I addressed the comments.

About the 2 bigger problems:

1. Fixed. I introduced that by mistake just before committing :facepalm: Thanks for spotting it!
2. Good point. I didn't change the previous implementation, but that number of branches is wrong. Would it work if I open a follow-up issue to tackle immediately? I prefer not to fix it here to keep things separate (I will move `getBranches` in the new project coordinator and that requires minor adjustments to the notebooks components as well)

Preview updated at https://lorenzotest.dev.renku.ch/

lorenzo-cavazzi commented:
I addressed the comments.

About the 2 bigger problems:

1. Fixed. I introduced that by mistake just before committing :facepalm: Thanks for spotting it!
2. Fixed. I didn't change the previous implementation, but that number of branches is indeed wrong. Now the branches are fetched once at project load. Mind that the number already excludes the autosaved branches.
We could further improve the logic to refresh periodically as we do with the other stats and commits, but that will be much simpler after migrating the function to the new project coordinator. I wouldn't include that in this PR.

Preview updated at https://lorenzotest.dev.renku.ch/

ciyer reviewed (approved):
Very nice fix!
